core/cont/inc/THashList.h

```diff
@@ -36,7 +36,7 @@ class THashTable;
 class THashList : public TList {
 
 protected:
-   THashTable   *fTable;    //Hashtable used for quick lookup of objects
+   THashTable   *fTable;    //! Hashtable used for quick lookup of objects
```

Comment:
Dear Philippe, 
technically, it's of course redundant - this change is to clarify that it is the class-author's intent that this member should be transient in a more explicit manner than from the class-version. 

This allows to distinguish these two cases programmatically: 
- Class with class-version 0 since it is not meant for IO, and should not be inherited from by other classes wich class-version >0. Example: `TGedFrame`. This GUI class does not need IO, and there is no sensible case in which a class inheriting from it should have IO / class-version > 0, since functionality will break. 
- Class with class-version 0 to make all realdata-members transient - but class is intended to be used for IO (e.g. as a baseclass). That's right now (if I found all cases) only `TSeqCollection` and `THashList` in ROOT. 

There is currently no special markup / keyword to distinguish these two cases in a programmatic way. 

So what my test is doing ( https://github.com/olifre/rootStaticAnalyzer/blob/master/src/tests/testDataObjBases.cpp ) is the following: 
- Walk through all realdata-members of all classes with class-version > 0. 
- Blame all non-transient realdata which is part of a class with class-version 0. 

Members "lost" this way in streaming are in > 90 % of the cases not due to user's intent, but since the class-writer did not understand the implicit transientness of all members due to class-version 0 (or just copy-pasted a `ClassDef` statement from somewhere else). I encounter these cases almost daily in framework development, and that's also how I found all these cases which are part of this PR, so this kind of test has proven to identify many really-unwanted cases of adding a Streamer to a class. 

However, this test is giving me two false positives: `THashList` and `TSeqCollection`. For these classes, the class-writer apparently on-purpose chose class-version 0 to make all members transient. But there is no programmatic way to distinguish this good case from the "bad", unintended one. The simplest way to clarify programmatically that this transientness is wanted is to add `//!`. 

So I make use of the 'faithfulness' of the information to highlight the cases when the user very likely did not realize the information is transient by asking the user to explicitly mark the few cases when the transientness for a member is really wanted (which for ROOT are just these two classes I believe). 

So yes, you are fully right, from the technical point of view, this is redundant - but this markup expresses user's intent and allows for programmatic testing of the codebase. 

That's also why I fully agree with you that the realdata should *not* have the transient-bit set if class-version is 0, but only if the member is explicitly marked transient (via `//!`). After all, my test relies on this faithfulness ;-). 

Cheers, 
Oliver